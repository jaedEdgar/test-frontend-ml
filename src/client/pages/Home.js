import React from 'react';
import Navbar from '../components/navbar';
import useSearch from "../hooks/useSearch"

export default function Home() {
  const [valueSearch, setValueSearch, handleSubmitSearch] = useSearch()
  return (
    <Navbar
      handleSubmitSearch={handleSubmitSearch}
      value={valueSearch}
      handleValueChange={setValueSearch}
    />
  );
}
