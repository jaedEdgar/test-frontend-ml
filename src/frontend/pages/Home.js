import React, { useState } from 'react';
import Navbar from '../components/navbar';
import { useHistory } from 'react-router-dom';

export default function Home() {
  let history = useHistory();
  const [valueSearch, setValueSearch] = useState('');

  const handleSubmitSearch = () => {
    history.push(`/items?search=${encodeURI(valueSearch)}`);
  };

  return (
    <>
      <Navbar
        handleSubmitSearch={() => handleSubmitSearch()}
        value={valueSearch}
        handleValueChange={setValueSearch}
      />
    </>
  );
}
