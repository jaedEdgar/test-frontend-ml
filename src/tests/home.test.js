import React from 'react';
import renderer from 'react-test-renderer';
import Home from "../client/pages/Home"
import '@testing-library/jest-dom'

test('should render my component', () => {
  const component = renderer.create(<Home />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
