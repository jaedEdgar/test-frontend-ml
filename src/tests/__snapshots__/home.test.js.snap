// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render my component 1`] = `
<header
  className="navbar navbar-search"
  onSubmit={[Function]}
>
  <div
    className="container"
  >
    <a
      className="logo"
      href="/"
    />
    <form
      className="navbar-search--form"
    >
      <input
        className="navbar-search--input"
        name="search"
        onChange={[Function]}
        placeholder="Nunca dejes de buscar"
        type="text"
        value=""
      />
      <button
        className="navbar-search--button"
        onClick={[Function]}
        type="submit"
      >
        <span />
      </button>
    </form>
  </div>
</header>
`;
